httpsupport/url.go

```diff
@@ -1,18 +1,36 @@
 package httpsupport
 
 import (
-	"bytes"
 	"fmt"
-	"io"
-	"io/ioutil"
-	"net/http"
 	"strings"
 
 	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/goadesign/goa"
 )
 
 // AbsoluteURL prefixes a relative URL with absolute address
-func AbsoluteURL(req *http.Request, relative string) string {
+// If config is not nil and run in dev mode then host is replaced by "auth.openshift.io"
+func AbsoluteURL(req *goa.RequestData, relative string, config configuration) string {
+	host := Host(req, config)
+	return absoluteURLForHost(req, host, relative)
+}
+
+// ReplaceDomainPrefixInAbsoluteURL replaces the last name in the host of the URL by a new name.
+// Example: https://api.service.domain.org -> https://sso.service.domain.org
+// If replaceBy == "" then return trim the last name.
+// Example: https://api.service.domain.org -> https://service.domain.org
+// Also prefixes a relative URL with absolute address
+// If config is not nil and run in dev mode then "auth.openshift.io" is used as a host
+func ReplaceDomainPrefixInAbsoluteURL(req *goa.RequestData, replaceBy, relative string, config configuration) (string, error) {
+	host := Host(req, config)
+	newHost, err := ReplaceDomainPrefix(host, replaceBy)
+	if err != nil {
+		return "", err
+	}
+	return absoluteURLForHost(req, newHost, relative), nil
+}
```

Comment:
Should tests be ultimate example, documentation of the API in use? :) The `_test.go` you mention does punch of prints, but doesn't really validate anything

Reply:
exampe_test.go is not a test. It’s example. Usually tests are too verbose to be a good example.
I guess they use _test prefix for examples just to avoid packaging it into prod binary.